amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/parser/document/JsonSchemaLinker.scala

```diff
@@ -18,29 +18,40 @@ import amf.shapes.internal.validation.definitions.ShapeParserSideValidations.{
 import org.yaml.model.YNode
 
 trait NameExtraction {
-  def extract(ref: String): Either[String, String]
+  def extract(ref: String, declarationKey: Option[String]): Either[String, String]
 }
 
 trait JsonSchemaRefNameExtraction extends NameExtraction {
   protected def extractShapeName(uriFragment: String, definitionsKey: String): Option[String] = {
     val regex = ("^\\/" + definitionsKey + "\\/(?<shape>[^\\/]+)$").r
     regex.findFirstMatchIn(uriFragment).map(_.group(1))
   }
+
+  protected def validateDeclarationKey(
```

Comment:
Can't this be out of here and inside the `findShapeForReference` function ?

Reply:
My problem with this is that we have to remember to call it on each NameExtraction. We validate the correct ref format ($defs or definitions) before calling name extraction

Reply:
I did that first, but the problem is that doing that I return always the same error, and in some cases is not specific enough. Putting it here it checks first at least that the link is a correctly formatted declaration link.